File: hlmng/src/hlmng/model/Media.java
package hlmng.model;

public class Media {

	private int mediaID;
	private String type;
	private String link;
	
	public Media(){
		
	}
	public Media(int mediaID, String type, String link){
		this.setMediaID(mediaID);
		this.setType(type);
		this.setLink(link);
	}
	public Media(String type, String link){
		this.setType(type);
		this.setLink(link);
	}
	
	public String getType() {
		return type;
	}
	public void setType(String type) {
		this.type = type;
	}
	public String getLink() {
		return link;
	}
	public void setLink(String link) {
		this.link = link;
	}
	public int getMediaID() {
		return mediaID;
	}
	public void setMediaID(int mediaID) {
		this.mediaID = mediaID;
	}
	
}

File: hlmng/WebContent/js/media.js
var mediaModule = angular.module('media', []);


app.controller('MediaController', ['$scope', 'ToolService','$log', function($scope, ToolService,$log){
	var hlmng = this;
	$scope.uploadedMediaLink={};
	$scope.uploadedMediaID={};
	hlmng.redir=ToolService.redir;
	$scope.uploadFile = function(file){
		$log.log("Starting upload in media controller");
        ToolService.uploadFile(file).then(function(data){
        	$scope.uploadedMediaLink=data.link;
        	$scope.uploadedMediaID=data.mediaID;
        });
    };
}]);


mediaModule.directive('mediaUpload', function(){
	return {
		restrict: 'E',
		templateUrl: 'template/media/upload.html',
		scope:{
		}
	};
});

	
// This code comes from: https://uncorkedstudios.com/blog/multipartformdata-file-upload-with-angularjs
mediaModule.directive('fileModel', ['$parse', function ($parse) {
    return {
        restrict: 'A',
        link: function(scope, element, attrs) {
            var model = $parse(attrs.fileModel);
            var modelSetter = model.assign;
            
            element.bind('change', function(){
                scope.$apply(function(){
                    modelSetter(scope, element[0].files[0]);
                });
            });
        }
    };
}]);